### PR TITLE
fix: Fix evaluation logic

### PR DESCRIPTION
**Pull request by robinholzi**
# Motivation

There were some issues that evaluation results weren't collected into the logfile.

### CODE REVIEW

File: modyn/trainer_server/internal/trainer/pytorch_trainer.py [hunk 1]
```diff
@@ -376,7 +375,6 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
 
         self._info(f"Finished training: {self._num_samples} samples, {batch_number + 1} batches.")
         self._log["num_samples"] = self._num_samples
-        self._log["num_samples_trained"] = trained_batches * self._batch_size
```
Comment (robinholzi) [inline modyn/trainer_server/internal/trainer/pytorch_trainer.py:379]: Looks like a auto-merge issue, we need this, right?

File: modyn/supervisor/internal/pipeline_executor/evaluation_executor.py [hunk 2]
```diff
@@ -103,12 +103,14 @@ def init_from_path(cls, pipeline_logdir: Path) -> "EvaluationExecutor":
         eval_state_config = EvalStateConfig.model_validate_json((snapshot_dir / "eval_state.yaml").read_text())
         context = pickle.loads((snapshot_dir / "context.pcl").read_bytes())
 
+        grpc_handler = GRPCHandler(eval_state_config.config.model_dump(by_alias=True))
+        grpc_handler.init_cluster_connection()
```
Comment (MaxiBoether) [inline modyn/supervisor/internal/pipeline_executor/evaluation_executor.py:107]: Haha, I also just fixed that locally :D

Comment (robinholzi) [reply modyn/supervisor/internal/pipeline_executor/evaluation_executor.py:107]: had it in my other branch for some time now, but this seems to be a good place to sneak it in :)

**Review by MaxiBoether — APPROVED** [review]
Thanks. As asked on element, you mentioned before that you might add an increment log in _done, but not sure if that makes sense or not.

**Pull request merged**